### PR TITLE
Expand parquet docs, multiple files docs, and add hive partitioning/partitioned writes docs

### PR DESCRIPTION
**Pull request by Mytherin**
* This PR splits up the Parquet import docs into separate pages (`overview`, `metadata`, `tips`).
* Reading multiple files is split up into two separate docs pages (`Multiple Files` and `Combining Schemas`) - as these are related to CSV, JSON and Parquet ingestion
* Documentation is added on hive partitioning as well as partitioned writes (`Hive Partitioning`, `Partitioned Writes`), CC @samansmink
* The `COPY` documentation page is cleaned up, and the parameters are split into parameters for the separate formats (general, `CSV`, `Parquet` and `JSON`).

In addition I have converted the parameters to lowercase - which just looks a lot better/more readable in my opinion.

### CODE REVIEW

File: docs/data/multiple_files/overview.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+SELECT * FROM read_parquet(['file1.parquet', 'file2.parquet', 'file3.parquet']);
+```
+
+#### Glob Syntax
+Any file name input to the read_parquet function can either be an exact filename, or use a glob syntax to read multple files that match a pattern.
+
+|  Wildcard  |                        Description                        |
+|------------|-----------------------------------------------------------|
+| `*`        | matches any number of any characters (including none)     |
+| `?`        | matches any single character                              |
```
Comment (samansmink) [inline docs/data/multiple_files/overview.md:52]: Maybe we should mention here that we recently disabled the ? in globs for S3 urls

**Review by samansmink — COMMENTED** [review]
great idea to improve on this, added one small comment

File: docs/data/multiple_files/overview.md (hunk 1 above)
Comment (Mytherin) [reply docs/data/multiple_files/overview.md:52]: Added a note, let me know if you want to change anything about it

Comment (samansmink) [reply docs/data/multiple_files/overview.md:52]: perfect!

File: docs/data/partitioning/hive_partitioning.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+-- read data from a hive partitioned data set
+SELECT * FROM parquet_scan('orders/*/*/*.parquet', hive_partitioning=1);
+-- write a table to a hive partitioned data set
+COPY orders TO 'orders' (FORMAT PARQUET, PARTITION_BY (year, month));
+```
+
+### Hive Partitioning
+Hive partitionining is a [partitioning strategy](https://en.wikipedia.org/wiki/Partition_(database)) that is used to split a table into multiple files based on **partition keys**. The files are organized into folders. Within each folder, the **partition key** has a value that is determined by the name of the folder. 
+
+Below is an example of a hive partitioned file hierarchy. The files are partitioned on two keys (`year` and `order`).
```
Comment (Alex-Monahan) [inline docs/data/partitioning/hive_partitioning.md:18]: Small typo: `order` should be `month`

**Comment by Alex-Monahan** [comment]
Thanks @Mytherin, this is fantastic! I added a few tips to the Parquet tips page and added in a note about boolean parameters to the copy option. I think this looks good to go!

**Pull request merged**